src/planscape/planning/serializers.py

```diff
@@ -389,11 +390,24 @@ class Meta:
         )
 
 
+class UploadedConfigurationSerializer(serializers.Serializer):
+    stand_size = serializers.ChoiceField(choices=StandSizeChoices.choices)
+
+
 class ScenarioSerializer(
     ListScenarioSerializer,
     serializers.ModelSerializer,
 ):
-    configuration = ConfigurationSerializer()
+    configuration = serializers.SerializerMethodField()
+
+    def __init__(self, *args, **kwargs):
+        instance = kwargs.get("instance", None)
+
+        super().__init__(*args, **kwargs)
+        if instance and hasattr(instance, "origin") and instance.origin == "SYSTEM":
+            self.fields["configuration"] = ConfigurationSerializer()
+        else:
+            self.fields["configuration"] = UploadedConfigurationSerializer()
```

Comment:
All scenarios have `configuration` validated in various places, so making this dynamic seemed like the only real option. I'm not sure if there's a better way to do this, but this solves the issue I was seeing.